jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -356,13 +356,16 @@ protected void startContext() throws Exception
 
         if (_servletHandler != null)
         {
-            // Call decorators on all holders, and also on any EventListeners before
-            // decorators are called on any other classes (like servlets and filters)
+            //Ensure listener instances are created, added to ContextHandler
             if(_servletHandler.getListeners() != null)
             {
                 for (ListenerHolder holder:_servletHandler.getListeners())
                 {             
-                    _objFactory.decorate(holder.getListener());
+                    holder.start();
```

Comment:
Are holders stopped and/or destroyed with the context?

Reply:
Holders are stopped when the ServletHandler is stopped (same as ServletHolders and FilterHolders). Listener instances are cleared out on the ContextHandler when the context stops.